streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java

```diff
@@ -109,7 +109,11 @@ static void afterClassGlobalCleanup() {
     void beforePrepareTest() throws Exception {
         ++testNo;
         bootstrapServers = cluster.bootstrapServers();
-        mockTime = cluster.time;
```

Comment:
Shouldn't we sync broker mock time ? IMHO, in a test, there should be only one mock time object that is used globally. With this change, we get an internal broker time and a external test mock time and both are decoupled.

Reply:
The problem is that, the same mock time object will be used for more than one test method within the class, so for the second test method the time is not aligned.

Reply:
I agree. But can't we align it differently than? In before test, we get mockTime value and "tick" as many times as needed to align?

Reply:
Could we add a `setTime` method to `MockTime` to re-align it? i.e., do it in the `Before` method?

Reply:
Sounds good. Will do.